framework/src/utils/PetscSupport.C

```diff
@@ -733,6 +737,63 @@ setSinglePetscOption(const std::string & name, const std::string & value)
     mooseError("Error setting PETSc option.");
 }
 
+void
+colorAdjacencyMatrix(PetscScalar * adjacency_matrix,
+                     unsigned int size,
+                     unsigned int colors,
+                     std::vector<unsigned int> & vertex_colors,
+                     const char * coloring_algorithm)
+{
+  // Mat A will be a dense matrix from the incoming data structure
+  Mat A;
+  MatCreate(MPI_COMM_SELF, &A);
+  MatSetSizes(A, size, size, size, size);
+  MatSetType(A, MATSEQDENSE);
+  // PETSc requires a non-const data array to populate the matrix
+  MatSeqDenseSetPreallocation(A, adjacency_matrix);
+  MatAssemblyBegin(A, MAT_FINAL_ASSEMBLY);
+  MatAssemblyEnd(A, MAT_FINAL_ASSEMBLY);
+
+  // Sparse matrix
+  Mat B;
+  MatConvert(A, MATAIJ, MAT_INITIAL_MATRIX, &B);
```

Comment:
It should be very easy to construct a sparse matrix based on the original data. There is no need to build a temporary dense matrix.